pkgs/development/compilers/llvm/10/llvm.nix

```diff
@@ -100,6 +100,9 @@ in stdenv.mkDerivation (rec {
     ln -sv $PWD/lib $out
   '';
 
+  # mesa drivers use the build-id as a cache key
+  LDFLAGS = optionalString enableSharedLibraries "-Wl,--build-id=sha1";
```

Comment:
Lack of build-id in llvm isn't likely to cause problems.  As you mentioned, it would be pretty difficult to use a different llvm without changing the build-id of mesa (absolute path to llvm is in RPATH).  The only way I can imagine this helping is if you're forcing it to load a different llvm with LD_ vars or something...

That being said, the timestamp fallback will always be broken on NixOS, so I can imagine a situation where we'd want to print an error when it's hit, and this would help in that case.  It also just feels like the right thing to do.

Again, we can always just merge a23c6888d399b291ba691e43028e1b295ad10efc if there's a concern.